cypress/e2e/buy-shirt.cy.ts

```diff
@@ -1,26 +1,40 @@
+import {MenuContentPage} from "../page/index";
+import {LoginPage} from "../page/index";
+import {AddressStepPage} from "../page/index";
+import {PaymentStepPage} from "../page/index";
+import {ProductsListPage} from "../page/index";
+import {ShippingStepPage} from "../page/index";
+import {ShopingCartPage} from "../page/index";
+
+const menuContentPage = new MenuContentPage();
+const loginPage = new LoginPage();
+const addressStepPage = new AddressStepPage();
+const paymentStepPage = new PaymentStepPage();
+const productsListPage = new ProductsListPage();
+const shippingStepPage = new ShippingStepPage();
+const shopingCartPage = new ShopingCartPage();
+
 describe("Buy a t-shirt", () => {
-  it("then the t-shirt should be bought", () => {
-    cy.visit("http://automationpractice.com/");
-    cy.get("#block_top_menu > ul > li:nth-child(3) > a").click();
-    cy.get(
-        "#center_column a.button.ajax_add_to_cart_button.btn.btn-default",
-    ).click();
-    cy.get("[style*='display: block;'] .button-container > a").click();
-    cy.get(".cart_navigation span").click();
-
-    cy.get("#email").type("aperdomobo@gmail.com");
-    cy.get("#passwd").type("WorkshopProtractor");
-
-    cy.get("#SubmitLogin").click();
-    cy.get("[name='processAddress']").click();
-    cy.get("#cgv").click();
-    cy.get("[name='processCarrier']").click();
-    cy.get(".bankwire").click();
-    cy.get("#cart_navigation > [type='submit']").click();
-
-    cy.get("#center_column > div > p > strong").should(
-        "have.text",
-        "Your order on My Store is complete.",
-    );
+  it("then should be bought a t-shirt", () => {
+    menuContentPage.visitMenuContentPage();
+    menuContentPage.goToTShirtMenu();
+
+    shopingCartPage.addShoppingCart();
+    shopingCartPage.clickButtonContainer();
+    shopingCartPage.toLogin();
+
+    loginPage.putEmailUser();
+    loginPage.putPasswordUser();
+    loginPage.goToAddressCheckout();
+
+    addressStepPage.addressCheckout();
+
+    shippingStepPage.checkAgreement();
+    shippingStepPage.goToCarrier();
+
+    paymentStepPage.bankWireclick();
+
+    productsListPage.clickOncheckout();
+    productsListPage.checkOrderFinal();
```

Comment:
rename the method as verifyConfirmationMessage or verifyFinalOrderMessage, pass the method to verify as a parameter of the method, and move the method to the paymentStepPage

Reply:
Hi, I don't get clear the part of pass the method to verify as a parameter of the method.
So I put te cy.get(this.checkOrder).should(this.checkText, this.checkOrderOnStore); as VerifyfinalOrderMessage(cy.get(this.checkOrder).should(this.checkText, this.checkOrderOnStore));?

Reply:
we talked about this in the last two sessions. You have two choices, implement the one you feel better:

1. Change the method to be named as getConfirmationTitle or something. That method should return the confirmation title, then you can perform the assertion here in this file (buy-shirt.cy.ts): `paymentStepPage.getConfirmationTitle().should("have.text", "Your order on My Store is complete.")`
2. Change the method to be named as verifyConfirmationTitle, that method should receive the message to validate, and then, within that method on the paymentStepPage, you should perform the assertion. So, here in this file we should see something like: `paymentStepPage.verifyConfirmationTitle("Your order on My Store is complete.")`